src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -133,8 +133,12 @@ private static CreationResultStatus DisplayHelpForAmbiguousTemplateGroup(Templat
                 DisplayParametersInvalidForSomeTemplates(invalidForSomeTemplates, LocalizableStrings.PartialTemplateMatchSwitchesNotValidForAllMatches);
             }
 
-            ShowContextAndTemplateNameMismatchHelp(templateResolutionResult, commandInput.TemplateName, commandInput.TypeFilter);
-            DisplayTemplateList(templatesForDisplay, environmentSettings, commandInput.Language, defaultLanguage);
+            bool hasTemplateList;
+            ShowContextAndTemplateNameMismatchHelp(templateResolutionResult, commandInput.TemplateName, commandInput.TypeFilter, out hasTemplateList);
```

Comment:
I have 2 minor comments here:
1. `hasTemplateList` would better state it's purpose if we renamed it to `shouldListTemplates` or `shouldDisplayTemplateList` all the way down.
1. We can inline the declaration and get rid of the first line like: `... TypeFilter, out bool hasTemplateList);`